### DISCUSSION TITLE
Noob question about handling of thumbnails

### DISCUSSION THREAD

**Discussion by RodgeCF** [stashapp/stash#4365] [Q&A] [answered]
If I add a custom image for a performer or tag, does Stash copy those images to somewhere else, or just retrieve them from their original location?

**Comment by dingdongsolong4** [comment, marked as answer]
Stash stores copies of all images as "blobs" which are either stored in the database, if `Binary data storage type` (in `Settings > System > Database`) is set to `Database`,  or the `Binary data filesystem path`, if `Binary data storage type` is set to `Filesystem`.

If you add an image "From file", then you are uploading the image to the Stash server, and it will store that image as above. If you add an image "From URL", the Stash server will download the image from that URL, and then store it as above.
